combinetf2/inputdata.py

```diff
@@ -0,0 +1,212 @@
+
+            else:
+                self.channel_info = {
+                    "ch0": {
+                        "axes": [
+                            hist.axis.Integer(
+                                0,
+                                self.nbins,
+                                underflow=False,
+                                overflow=False,
+                                name="obs",
+                            )
+                        ]
+                    }
+                }
+
+            self.symmetric_tensor = self.metadata.get("symmetric_tensor", False)
+            self.exponential_transform = self.metadata.get(
+                "exponential_transform", False
+            )
+            self.exponential_transform_scale = self.metadata.get(
+                "exponential_transform_scale", 1000000
+            )
+
+            # compute indices for channels
+            ibin = 0
+            for channel, info in self.channel_info.items():
+                axes = info["axes"]
+                shape = tuple([len(a) for a in axes])
+                size = np.prod(shape)
+
+                start = ibin
+                stop = start + size
+
+                info["start"] = start
+                info["stop"] = stop
+
+                ibin = stop
+
+            for channel, info in self.channel_info.items():
+                print(channel, info)
+
+            self.axis_procs = hist.axis.StrCategory(self.procs, name="processes")
+
+            # build tensorflow graph for likelihood calculation
+
+            # start by creating tensors which read in the hdf5 arrays (optimized for memory consumption)
+            self.constraintweights = maketensor(hconstraintweights)
+
+            # load data/pseudodata
+            if pseudodata is not None:
+                if pseudodata in self.pseudodatanames:
+                    pseudodata_idx = np.where(self.pseudodatanames == pseudodata)[0][0]
+                else:
+                    raise Exception(
+                        "Pseudodata %s not found, available pseudodata sets are %s"
+                        % (pseudodata, self.pseudodatanames)
+                    )
+                print("Run pseudodata fit for index %i: " % (pseudodata_idx))
+                print(self.pseudodatanames[pseudodata_idx])
+                hdata_obs = f["hpseudodata"]
+
+                data_obs = maketensor(hdata_obs)
+                self.data_obs = data_obs[:, pseudodata_idx]
+            else:
+                self.data_obs = maketensor(hdata_obs)
+
+            hkstat = f["hkstat"]
+            self.kstat = maketensor(hkstat)
+
+            if self.sparse:
+                self.norm_sparse = makesparsetensor(hnorm_sparse)
+                self.logk_sparse = makesparsetensor(hlogk_sparse)
+            else:
+                self.norm = maketensor(hnorm)
+                self.logk = maketensor(hlogk)
+
+            self.normalize = normalize
+            if self.normalize:
```

Comment:
this would need to be adapted to support the symmetric_tensor case

suggest adding a NotImplementedError for this case for the moment at least